lib/bats-core/tracing.bash

```diff
@@ -72,18 +78,24 @@ bats_print_failed_command() {
 	local lineno
 	local failed_line
 	local failed_command
+  local output=''
 
 	bats_frame_filename "$frame" 'filename'
 	bats_frame_lineno "$frame" 'lineno'
 	bats_extract_line "$filename" "$lineno" 'failed_line'
 	bats_strip_string "$failed_line" 'failed_command'
-	printf '%s' "#   \`${failed_command}' "
+  output+=$(printf '%s' "#   \`${failed_command}' ")
 
 	if [[ "$BATS_ERROR_STATUS" -eq 1 ]]; then
-		printf 'failed\n'
+    output+="failed"
 	else
-		printf 'failed with status %d\n' "$BATS_ERROR_STATUS"
+    output+="failed with status $BATS_ERROR_STATUS"
 	fi
+  
+  echo "$output"
```

Comment:
Please use printf for all outputs in bats' bash files.

Reply:
> This is somewhat similar to the trace PR. I like the idea of having a separate log file but there may be situations where we warnt the output in the test result.
> 
> My suggestion would to have two options --trace and --trace-log to decide where the output goes.
> 
> Alternatively we could add FD 4 that is meant for trace output an can be redirected/merged at will. I'd have to think a bit about this.

Having this output to STDOUT of bats would be awesome, but I can't seem to get it working. This is why I just resorted to using a log file.